processing/src/main/java/org/apache/druid/segment/IndexIO.java

```diff
@@ -643,37 +649,38 @@ public QueryableIndex load(File inDir, ObjectMapper mapper, boolean lazy, Segmen
           smooshedFiles,
           loadFailed
       );
+      // todo (clint): just pass in the metadata to SimpleQueryableIndex so it doesnt need to read it again...
```

Comment:
The actual Metadata object you mean? That would increase heap pressure quite a bit, and is I think something we'd want to avoid.

Reply:
i don't mean to retain a reference to it, but we already read it in the constructor to populate ordering, so instead of doing that we should just pass it in so we don't have to deserialize it twice